tests/crc32/main.c

```diff
@@ -50,19 +143,53 @@ selftest(void)
 		} else {
 			printf(" PASSED!\n");
 		}
+
+		/* Prepare the buffer in non-aligned memory. */
+		printf("Computing CRC32C of \"%s\" with non-aligned "
+		    "memory...", tests[i].s);
+		if ((unaligned = malloc(strlen(tests[i].s) + 3)) == NULL)
+			goto err0;
+		memcpy(unaligned + 3, tests[i].s, strlen(tests[i].s));
+
+		/* Try again with unaligned memory. */
```

Comment:
I think a better test here would be to construct a large buffer "helloworldhelloworldhelloworldhelloworld..." and then (a) hash it as a single buffer, and (b) feed in `random() % 256` bytes at a time.

That should ensure that we get every combination of lengths mod 8 and offsets mod 8, as well as testing that a series of Update calls does what they should -- right now I think we're only ever testing with a single Update call before we invoke Final.